tests/func.sh

```diff
@@ -357,14 +357,35 @@ check() {
 		max=`cat /proc/sys/dev/raid/speed_limit_max`
 		echo 200000 > /proc/sys/dev/raid/speed_limit_max
 		sleep 0.1
+		max_iterations=10
+		# Wait if one of the actions appears in sync_action.
 		while grep -Eq '(resync|recovery|reshape|check|repair) *=' /proc/mdstat ||
 			grep -v idle > /dev/null /sys/block/md*/md/sync_action
 		do
-			sleep 0.5
+			if (( max_iterations > 0 )); then
+				sleep 0.5
+				((max_iterations--))
+			else
+				break
+			fi
 		done
-		while ps auxf | grep "mdadm --grow --continue" | grep -v grep
+
+		# wait for grow-continue to finish but break if sync_action does not
+		# contain any reshape value
+		while pgrep -f "mdadm --grow --continue" > /dev/null
```

Comment:
Please move this to separate function because the block is not readable to me.

Caller should tell us that we should verify reshape instead of optimistically looping for it. Maybe it is not an issue now, but direct parameter is better readable (you always know when this code is executed).

Reply:
Now it depends on your knowledge in the raid area, I want to lower the complexity of jumping into development.